src/main.rs

```diff
@@ -1140,31 +1145,118 @@ fn cmd_suggest(
     Ok(())
 }
 
-fn cmd_regenerate_unaudited(
+fn cmd_regenerate_imports(
+    out: &mut dyn Out,
+    cfg: &Config,
+    _sub_args: &RegenerateImportsArgs,
+) -> Result<(), miette::Report> {
+    trace!("regenerating imports...");
+
+    let mut store = Store::acquire(cfg)?;
+    let network = Network::acquire(cfg);
+
+    if let Some(network) = &network {
+        if !cfg.cli.locked {
+            // Literally the only difference between this command and fetch-imports
+            // is that we pass `accept_changes = true`
+            tokio::runtime::Handle::current()
+                .block_on(store.fetch_foreign_audits(network, true))?;
+            store.commit()?;
+            return Ok(());
+        }
+    }
+
+    // ERRORS: just a warning that you're holding it wrong, unclear if immediate or buffered,
+    // or if this should be a hard error, or if we should ignore the --locked flag and
+    // just do it anyway
+    writeln!(
+        out,
+        "warning: ran `regenerate imports` with --locked, this won't do anything!"
+    )
+    .into_diagnostic()?;
```

Comment:
Should this perhaps be a `warn!` through tracing? I suppose `fetch-imports` does it this way as well.

Reply:
Yeah I'm just reproducing the existing issue.